astropy/io/fits/tests/test_connect.py

```diff
@@ -43,6 +43,20 @@ def test_simple(self, tmpdir):
         t2 = Table.read(filename)
         assert equal_data(t1, t2)
 
+    def test_append_mode_is_determined_correctly(self, tmpdir):
+        # TODO: Move this test in test_util as soon as I figure out how to
+        # setup a class with tmpdir...
```

Comment:
I use temporary directory in `astropy/vo/validator/tests/test_validate.py`. Something like
```python
import shutil
import tempfile

class TestConeSearchValidation(object):
    def setup_class(self):
        self.out_dir = tempfile.mkdtemp()
    def teardown_class(self):
        shutil.rmtree(self.out_dir)
```

Is this what you are looking for?